data/hydraulic-data.py

```diff
@@ -0,0 +1,129 @@
+                pd_targetLabelY = pd.DataFrame(pd_frame.iloc[:,1])
+            elif target_label == "pump_leak_y":
+                # store hydraulic internal pump leakage label data from column 2 into pd frame
+                pd_targetLabelY = pd.DataFrame(pd_frame.iloc[:,2])
+            elif target_label == "acc_gas_leak_y":
+                # store hydraulic accumulator gas leakage label data from column 3 into pd frame
+                pd_targetLabelY = pd.DataFrame(pd_frame.iloc[:,3])
+            elif target_label == "stable_y":
+                # store hydraulic stable flag label data from column 4 into pd frame
+                pd_targetLabelY = pd.DataFrame(pd_frame.iloc[:,4])
+            else:
+                sys.exit("Not valid hydraulic condition label")
+            pd_targetLabelY.columns = [target_label]
+            return pd_targetLabelY
+
+        def extract_hydraulic_zip(file, output_file, target_label):
+            with zipfile.ZipFile(file,"r") as zip:
+                # import all data of pressure sensors
+                df_ps1 = import_data_field(zip, "PS1.txt", "psa_bar")
+                df_ps2 = import_data_field(zip, "PS2.txt", "psb_bar")
+                df_ps3 = import_data_field(zip, "PS3.txt", "psc_bar")        
+                df_ps4 = import_data_field(zip, "PS4.txt", "psd_bar")
+                df_ps5 = import_data_field(zip, "PS5.txt", "pse_bar")
+                df_ps6 = import_data_field(zip, "PS6.txt", "psf_bar")
+
+                # import all data of volume flow sensors
+                df_vf1 = import_data_field(zip, "FS1.txt", "fsa_vol_flow")
+                df_vf2 = import_data_field(zip, "FS2.txt", "fsb_vol_flow")
+
+                # import all data of temperature sensors
+                df_ts1 = import_data_field(zip, "TS1.txt", "tsa_temp")
+                df_ts2 = import_data_field(zip, "TS2.txt", "tsb_temp")
+                df_ts3 = import_data_field(zip, "TS3.txt", "tsc_temp")
+                df_ts4 = import_data_field(zip, "TS4.txt", "tsd_temp")
+
+                # import data of pump efficiency sensor
+                df_pe1 = import_data_field(zip, "EPS1.txt", "pump_eff")
+
+                # import data of vibrations sensor
+                df_vs1 = import_data_field(zip, "VS1.txt", "vs_vib")
+
+                # import data of cooling efficiency sensor
+                df_ce1 = import_data_field(zip, "CE.txt", "cool_eff_pct")
+
+                # import data of cooling power sensor
+                df_cp1 = import_data_field(zip, "CP.txt", "cool_pwr_pct")
+
+                # import data of efficiency factor sensor
+                df_ef1 = import_data_field(zip, "SE.txt", "eff_fact_pct")
+
+                # import one of the hydraulic condition monitoring labels from profile based on target label
+                df_hydCondMonitorLabelY = import_target_label(zip, "profile.txt", target_label)        
```

Comment:
nit: Regarding naming styles (`lowercase_with_underscores` vs `df_mixedCase`), different cases are used in this script.  Maybe it would make sense to stick with one or the other?  For reference: https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles

Reply:
I decided to go with lowercase_with_underscores for all custom variable names and function names used in the code.